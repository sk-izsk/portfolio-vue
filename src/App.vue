<template>
  <div id="app">
    <Home />
  </div>
</template>

<script>
import Home from "./screens/Home";
export default {
  name: "App",
  components: {
    Home,
  },
};
</script>

<style></style>
