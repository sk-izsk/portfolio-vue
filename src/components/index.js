import AboutMeCard from './AboutMeCard';
import FormContainer from './FormContainer';
import InfoCard from './InfoCard';
import InfoCardList from './InfoCardList';
import Modal from './Modal';
import NavBar from './NavBar';
import ProgressContainer from './ProgressContainer';
import Spinner from './Spinner';

export { NavBar, AboutMeCard, ProgressContainer, Spinner, InfoCard, InfoCardList, FormContainer, Modal };
