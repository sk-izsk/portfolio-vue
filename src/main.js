import { BootstrapVue, BootstrapVueIcons } from 'bootstrap-vue';
import 'bootstrap-vue/dist/bootstrap-vue.css';
import 'bootstrap/dist/css/bootstrap.css';
import Vue from 'vue';
import feather from 'vue-icon';
import VueScrollTo from 'vue-scrollto';
import App from './App.vue';
import store from './store';

Vue.use(VueScrollTo);
Vue.use(feather, 'v-icon');
Vue.use(BootstrapVue);
Vue.use(BootstrapVueIcons);

Vue.config.productionTip = false;

new Vue({
  store,
  render: (h) => h(App),
}).$mount('#app');
