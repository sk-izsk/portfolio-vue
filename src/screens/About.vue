<template>
  <div class="about" id="about">
    <h1>About Me</h1>
    <div class="info-prog-bar-container">
      <div class="info-container">
        <AboutMeCard v-if="url.length > 0" :url="url" />
      </div>
      <div class="progress-bar-container">
        <ProgressContainer v-if="skilsInformations.length > 0" :skilsInformations="skilsInformations" />
      </div>
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex';
import { AboutMeCard, ProgressContainer } from '../components/';
export default {
  name: 'About',
  components: {
    AboutMeCard,
    ProgressContainer,
  },
  data() {
    return {
      url: '',
      skilsInformations: [],
      variants: ['primary', 'secondary', 'info', 'success', 'danger'],
    };
  },
  computed: {
    ...mapState(['informations']),
  },
  beforeMount() {
    const { avatar, skillsInformation } = this.informations;
    if (avatar !== undefined && skillsInformation) {
      this.url = avatar.avatarTwo;
      this.skilsInformations = skillsInformation.map((skillInformation, index) => {
        return { ...skillInformation, variant: this.variants[index] };
      });
    }
  },
};
</script>

<style scoped>
.about {
  text-align: center;
  margin-top: 10px;
}
.info-prog-bar-container {
  display: flex;
}
.info-container {
  display: flex;
  flex: 1;
  padding: 20px;
  align-items: center;
  justify-content: center;
}
.progress-bar-container {
  display: flex;
  flex: 1;
  padding: 20px;
}
@media screen and (max-width: 874px) {
  .info-prog-bar-container {
    display: flex;
    flex-direction: column;
  }
}
</style>
