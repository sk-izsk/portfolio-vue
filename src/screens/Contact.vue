<template>
  <div id="contact">
    <h1>Contact</h1>
    <FormContainer />
  </div>
</template>

<script>
import { FormContainer } from '../components';
export default {
  name: 'Contact',
  components: {
    FormContainer,
  },
};
</script>

<style scoped>
#contact {
  display: flex;
  flex-direction: column;
  margin-bottom: 16px;
}
h1 {
  text-align: center;
  margin: 20px 0px;
}
</style>
