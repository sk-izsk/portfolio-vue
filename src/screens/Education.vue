<template>
  <div id="education">
    <Spinner variant="warning" v-if="educationInformations.length === 0" />
    <InfoCardList
      v-if="educationInformations.length > 0"
      :iconName="'book'"
      :title="'Education'"
      :informations="educationInformations"
    />
  </div>
</template>

<script>
import { InfoCardList, Spinner } from '../components';
import { mapState } from 'vuex';

export default {
  name: 'Education',
  components: {
    InfoCardList,
    Spinner,
  },
  data() {
    return {
      educationInformations: [],
    };
  },
  computed: {
    ...mapState(['informations']),
  },
  beforeMount() {
    const { educationInformations } = this.informations;
    if (educationInformations) {
      this.educationInformations = educationInformations;
    }
  },
};
</script>

<style scoped></style>
