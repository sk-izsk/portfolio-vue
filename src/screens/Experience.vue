<template>
  <div class="mt-3" id="experience">
    <Spinner variant="warning" v-if="experienceInformations.length === 0" />
    <InfoCardList
      v-if="experienceInformations.length > 0"
      :title="'Experience'"
      :informations="experienceInformations"
      :iconName="'bag-check'"
    />
  </div>
</template>

<script>
import { InfoCardList, Spinner } from '../components';
import { mapState } from 'vuex';

export default {
  name: 'Experience',
  components: {
    InfoCardList,
    Spinner,
  },
  data() {
    return {
      experienceInformations: [],
    };
  },
  computed: {
    ...mapState(['informations']),
  },
  beforeMount() {
    const { experienceInformations } = this.informations;
    if (experienceInformations) {
      this.experienceInformations = experienceInformations;
    }
  },
};
</script>

<style scoped></style>
