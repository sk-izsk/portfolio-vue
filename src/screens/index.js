import About from './About';
import Education from './Education';
import Experience from './Experience';
import Home from './Home';
import Photography from './Photography';
export { Home, About, Education, Experience, Photography };
