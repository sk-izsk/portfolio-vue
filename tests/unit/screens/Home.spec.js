import { shallowMount } from '@vue/test-utils';
import { Home } from '../../../src/screens/';

describe('Home', () => {
  it('renders when passed', () => {
    const wrapper = shallowMount(Home);
    expect(wrapper).toMatchSnapshot();
  });
});
