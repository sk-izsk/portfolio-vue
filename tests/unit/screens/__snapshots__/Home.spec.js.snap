// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders when passed 1`] = `
<div id="home" class="home">
  <div class="primary-text">Zeeshan</div>
  <div class="secondary-text">Full stack web developer</div>
  <div class="social-icons">
    <div>
      <v-icon name="github"></v-icon>
    </div>
    <div>
      <v-icon name="linkedin"></v-icon>
    </div>
    <div>
      <v-icon name="twitter"></v-icon>
    </div>
    <div>
      <v-icon name="facebook"></v-icon>
    </div>
    <div>
      <v-icon name="instagram"></v-icon>
    </div>
    <div>
      <v-icon name="send"></v-icon>
    </div>
    <div>
      <v-icon name="at-sign"></v-icon>
    </div>
  </div>
  <b-icon icon="arrow-down-circle" animation="throb" class="icon"></b-icon>
  <div class="mt-3 flex flex-column">
    <b-button pill="" size="lg" class="btn">Hire Me</b-button>
    <b-button pill="" size="lg" class="btn ml-2">React Portfolio</b-button>
  </div>
</div>
`;
